sycl/source/handler.cpp

```diff
@@ -577,11 +571,10 @@ event handler::finalize() {
         DiscardEvent = !KernelUsesAssert;
       }
 
-#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
-      if (!DiscardEvent) {
-        LastEventImpl = detail::event_impl::create_completed_host_event();
```

Comment:
No idea why we were using `create_completed_host_event` before. @slawekptak , any thoughts?